src/Xrefcheck/CLI.hs

```diff
@@ -69,6 +73,17 @@ addTraversalOptions TraversalConfig{..} (TraversalOptions ignored) =
   , ..
   }
 
+newtype VerifyOptions = VerifyOptions
+  { voIgnoreLocalhost :: Bool
+  }
+
+addVerifyOptions :: VerifyConfig -> VerifyOptions -> VerifyConfig
+addVerifyOptions VerifyConfig{..} (VerifyOptions ignoreLocalhost) =
+  VerifyConfig
+  { vIgnoreLocalhost = ignoreLocalhost
+  , ..
+  }
```

Comment:
Hm there seems to be a bug here.


1. If we have a `.xrefcheck.yaml` config file with `ignoreLocalhost: false`
2. and then call xrefcheck without any command line options

`xrefcheck` will still ignore localhost links, even though we've told it not to.

This happens because this function is overriding the `ignoreLocalhost: false` from the config file with the value parsed from the CLI (`--check-localhost`, which defaults to true when omitted).

Reply:
I've change it to `switch = flag False True`, so now default value is False

Reply:
Well, now you've reversed the problem :sweat_smile: 

1. If we have a `.xrefcheck.yaml` config file with `ignoreLocalhost: true`
2. and then call xrefcheck without any command line options

`xrefcheck` will still check localhost links, even though we've told it not to.

The problem is that `addVerifyOptions` _always_ ignores the value from the config, and uses only the value from the command line.

Reply:
(btw, I do agree with changing `--check-localhost` to `--ignore-localhost`, this way the naming matches the config option, so let's keep that!)

Reply:
> xrefcheck will still check localhost links, even though we've told it not to.

yes, but there are a question about priority - we have to define, which values more important for us - from config file or CLA. I think CLA has higher priority, because it requires some extra actions.

So, you have your config - and you just use it all the time. But than - there are situation where you pass some CLA - and you suppose, that your intention will be proceeded. 

What do you think?

Reply:
Yeah, I think CLA should have priority over config options. In the absence of a CLA, the config option should kick in.

Using my example from above, since no command line options were specified, and I had `ignoreLocalhost: true` in my config, I would have expected the config option to kick in.

Reply:
Let me put it another way.

The algorithm should be:

1. Use command line option if available
2. Otherwise, use config option if available
3. Otherwise, use some default value (in this case, I think the default is `ignoreLocalhost = false`)

However, with the current implementation:

1. Use command line option if available
2. Otherwise, use a default value of `false`

Reply:
oh, I've got your point, you are right

Reply:
@dcastro as I understand, 

> Use command line option if available
Otherwise, use config option if available
Otherwise, use some default value (in this case, I think the default is ignoreLocalhost = false)

we always have config option, because it doesn't wrapped with `Maybe`, so, I think the correct algorithm should be:
1. Use command line option if available
2. Otherwise, use config option

Reply:
@worm2fed Oh I didn't realize the config option was mandatory. Yup, that's correct then.

Reply:
By the way, the current implementation still doesn't work:

```
$ stack build xrefcheck:exe:xrefcheck && stack exec xrefcheck --  -r tests/markdowns/with-annotations/no_link.md                   
Missing: --ignore-localhost
```

From [`flag'`'s docs](https://hackage.haskell.org/package/optparse-applicative-0.16.1.0/docs/Options-Applicative-Builder.html#v:flag-39-):

> Builder for a flag parser without a default value.
> Same as flag, but with no default value. In particular, this flag will never parse successfully by itself.

At this point, I think we should do two things:

1. We should move the commits related to `ignoreAuthFailures` and `ignoreLocalhost` to a separate PR. They're holding up the merging of the FTP feature.
2. Looks like we could use some _golden tests_ to cover the use of command line options (in the new PR). These tests would call the `xrefcheck` executable just like a user would.

Reply:
Just noticed you've already created an issue specifically for the tests, #81.

Maybe we should amend the issue, so that it mentions both the implementation + the tests. I think we should also mention we need _golden_ tests.

Reply:
@dcastro make sense

Reply:
@worm2fed please ping me when the PR has been split and I'll approve it ^^

Reply:
@dcastro take a look at comments here please #81

Reply:
Cool, I'll have a look. In the meantime, can you remove those extra commits from this PR so we can merge it?

Reply:
Oh wait, I think I missed something. Do the FTP tests **require** the ability to check localhost links?

If so, does that mean that this PR depends on #85?

Reply:
> Oh wait, I think I missed something. Do the FTP tests require the ability to check localhost links?

yes) to run ftp tests we need to configure behaviour for localhost links

Reply:
> can you remove those extra commits from this PR 

I would like to do it when we'll merge #85 
cause at this commits there are some small changes I need to move to the main commit of this PR

Reply:
I see, ok, then I guess this PR is blocked until #85 is merged. Let's focus on #85 instead then :+1: